examples/next/docs/react/demo/src/renderers/ProgressBar.tsx

```diff
@@ -1,10 +1,41 @@
-import React from "react";
-import { addClassWhenNeeded, RendererProps } from "./utils";
+import React from 'react';
+import { baseRenderer } from 'handsontable/renderers';
+import { getRangeValue, HandsontableProps } from './utils';
+import { MESSAGE } from '../constants';
 
-export function ProgressBarRenderer(props: RendererProps) {
-  addClassWhenNeeded(props);
+const minAllowedValue = 0;
+const maxAllowedValue = 100;
+
+export const ProgressBarRenderer = (props: HandsontableProps) => {
+  const {
+    TD,
+    row,
+    col,
+    prop,
+    value,
+    cellProperties
+  } = props;
+  let isValid = cellProperties.valid;
+
+  baseRenderer(cellProperties.instance, TD, row, col, prop, value, cellProperties);
+
+  // Run the validator for the cell at initialization.
+  if (isValid === void 0) {
+    (cellProperties.validator as Function)(value, (isValueValid: boolean) => {
+      isValid = isValueValid;
+    });
+  }
```

Comment:
Maybe its worth to move up method, beyond the renderer declaration, to not repeat function creation and make it more clear by naming it. Suggestion up to you.